include/deal.II/base/exceptions.h

```diff
@@ -180,6 +180,12 @@ class ExceptionBase : public std::exception
 /**
  * Declare an exception class derived from ExceptionBase without parameters.
  *
+ * @note One can <code>#undef</code> this macro, as well as all other macros
```

Comment:
I think it would be worth adding a leading sentence of the form "This and similar class names are examples of preprocessor defines in the deal.II library that are not prefixed by a string that likely makes them unique to deal.II. As a consequence, it is possible that other libraries your code interfaces with defines similar names, and the result will then be name collisions (https://en.wikipedia.org/wiki/Name_collision). To resolve this issue, ..."